tower-buffer/src/error.rs

```diff
@@ -5,9 +5,9 @@ use std::sync::Arc;
 
 /// An error produced by a `Service` wrapped by a `Buffer`
 #[derive(Debug)]
-pub struct ServiceError<E> {
+pub struct ServiceError {
```

Comment:
should this impl clone now that it can?

Reply:
Generally, errors don't implement `Clone`, so i thought i would avoid it unless needed.